### PR TITLE
types for bootstrap operator

### PR DESCRIPTION
**Pull request by cornelius-keller**
@MarcelMue  can you have a quick look if what I did looks reasonable?

### CODE REVIEW

File: pkg/apis/bootstrap/v1alpha2/bootstrap_types.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type G8sBootsrapConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              G8sBootstrapSpec   `json:"spec"`
+	Status            G8sBootstrapStatus `json:"status"`
+}
+
+type G8sBootstrapSpec struct {
+	G8sVersion string `json:"g8sVersion,omitempty`
```
Comment (MarcelMue) [inline pkg/apis/bootstrap/v1alpha2/bootstrap_types.go]: json looks mangled

File: pkg/apis/bootstrap/v1alpha2/bootstrap_types.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+				Singular: "G8sBootstrapConfigconfig",
+			},
+		},
+	}
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type G8sBootsrapConfig struct {
+	metav1.TypeMeta   `json:",inline"`
```
Comment (MarcelMue) [inline pkg/apis/bootstrap/v1alpha2/bootstrap_types.go]: looks like something is missing in the json

File: pkg/apis/bootstrap/v1alpha2/bootstrap_types.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+				Plural:   "G8sBootstrapConfigconfigs",
+				Singular: "G8sBootstrapConfigconfig",
+			},
+		},
+	}
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type G8sBootsrapConfig struct {
```
Comment (MarcelMue) [inline pkg/apis/bootstrap/v1alpha2/bootstrap_types.go]:
```suggestion
type G8sBootstrapConfig struct {
```

File: pkg/apis/bootstrap/v1alpha2/bootstrap_types.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	// A slice of addresses ([]v1.NodeAddress) that contains a list of apiserver endpoints.
+	Addresses []v1.NodeAddress `json:"addresses,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type G8sBootstrapConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []G8sBootsrapConfig `json:"items"`
```
Comment (MarcelMue) [inline pkg/apis/bootstrap/v1alpha2/bootstrap_types.go]:
```suggestion
	Items           []G8sBootstrapConfig `json:"items"`
```

File: pkg/apis/bootstrap/v1alpha2/bootstrap_types.go (hunk 2 above)
Comment (cornelius-keller) [reply pkg/apis/bootstrap/v1alpha2/bootstrap_types.go]: copied from https://github.com/giantswarm/apiextensions/blob/master/pkg/apis/example/v1alpha1/memcached_types.go#L51 . How should it look like?

Comment (MarcelMue) [reply pkg/apis/bootstrap/v1alpha2/bootstrap_types.go]: You are correct, this is valid - my eyes were a bit too fast here :D

**Comment by xh3b4sd** [comment]
Is this PR still valid? Can we merge or close?

**Comment by cornelius-keller** [comment]
was only valid for operator week. Closing PR

**Pull request closed without merging**